<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Renee Matthews" />

<meta name="date" content="2025-05-29" />

<title>Glucose_counts_matrix</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Glucose_project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Glucose_counts_matrix</h1>
<h4 class="author">Renee Matthews</h4>
<h4 class="date">2025-05-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Glucose_project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250526code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250526)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250526code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250526)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereGlucoseprojecttree6d4d14fe4ddf2b66060c7e3919052ecd1b5345bftargetblank6d4d14fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Glucose_project/tree/6d4d14fe4ddf2b66060c7e3919052ecd1b5345bf" target="_blank">6d4d14f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereGlucoseprojecttree6d4d14fe4ddf2b66060c7e3919052ecd1b5345bftargetblank6d4d14fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Glucose_project/tree/6d4d14fe4ddf2b66060c7e3919052ecd1b5345bf" target="_blank">6d4d14f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/Annotation_data/
    Ignored:    data/Fastqc_data/
    Ignored:    data/organized_sample_info.tsv

Untracked files:
    Untracked:  code/nox_genelist.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Counts_matrix.Rmd</code>) and
HTML (<code>docs/Counts_matrix.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Glucose_project/blob/6d4d14fe4ddf2b66060c7e3919052ecd1b5345bf/analysis/Counts_matrix.Rmd" target="_blank">6d4d14f</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2025-05-30
</td>
<td>
updating PCA
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style type="text/css">
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
</style>
<pre class="r"><code>library(tidyverse)
library(readr)
library(kableExtra)
library(DT)
library(readr)
library(edgeR)
library(ComplexHeatmap)
library(data.table)
library(genomation)
library(GenomicRanges)
library(readxl)
library(stringr)
library(org.Hs.eg.db)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(RColorBrewer)
library(ggpubr)</code></pre>
<p>Data loading</p>
<pre class="r"><code>count_files &lt;- list.files(&quot;C:/Users/renee/Other_projects_data/Glucose_files/counts_files&quot;, pattern = &quot;\\.counts\\.txt$&quot;, full.names = TRUE)

counts_list &lt;- lapply(count_files, function(file) {
   df &lt;- read.delim(file, header = TRUE)
  
  # Rename &quot;X&quot; column to Geneid
  colnames(df)[1] &lt;- &quot;Geneid&quot;
  
  # Rename second column to a sample name (based on file name, not bam name)
  sample_name &lt;- tools::file_path_sans_ext(basename(file))
  colnames(df)[2] &lt;- sample_name
  
  # Keep only Geneid and the count column
  df &lt;- df[, c(&quot;Geneid&quot;, sample_name)]
  
  return(df)
})

merged_counts &lt;- Reduce(function(x, y) full_join(x, y, by = &quot;Geneid&quot;), counts_list)
merged_counts %&gt;%
  rename_with(~ str_remove(., &quot;_R1\\.counts$&quot;),
              .cols = -Geneid) %&gt;% 
  write_tsv(.,&quot;data/Annotation_data/Full_counts_table.tsv&quot;)</code></pre>
<pre class="r"><code>prop_var_percent &lt;- function(pca_result){ 
  # Ensure the input is a PCA result object
  if (!inherits(pca_result, &quot;prcomp&quot;)) {
    stop(&quot;Input must be a result from prcomp()&quot;)
  }
  
  # Get the standard deviations from the PCA result
  sdev &lt;- pca_result$sdev
  
  # Calculate the proportion of variance
  proportion_variance &lt;- (sdev^2) / sum(sdev^2)*100
  
  return(proportion_variance)
}</code></pre>
<pre class="r"><code>sample_info_concat &lt;- read_tsv(&quot;data/Annotation_data/organized_sample_info.tsv&quot;)
Counts_table &lt;- read_tsv(&quot;data/Annotation_data/Full_counts_table.tsv&quot;) 

raw_counts &lt;- Counts_table %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  as.matrix()
new_names &lt;-
sample_info_concat %&gt;% 
  mutate(new_trt= case_when(`Conc(mM)`==&quot;0&quot;~ &quot;low&quot;,
                            `Conc(mM)`==&quot;5&quot;~ &quot;norm&quot;,
                            `Conc(mM)`==&quot;5+25&quot; ~&quot;osnorm&quot;,
                            `Conc(mM)`==&quot;30&quot;~ &quot;high&quot;)) %&gt;% 
  mutate(sex=case_when(Line==&quot;75-1&quot;~ &quot;f&quot;,
                       Line==&quot;87-1&quot;~ &quot;f&quot;,
                       Line==&quot;78-1&quot; ~&quot;f&quot;,
                       Line==&quot;17-3&quot;~ &quot;m&quot;,
                       Line==&quot;84-1&quot;~&quot;m&quot;,
                       Line==&quot;90-1&quot;~&quot;m&quot;)) %&gt;% 
  mutate(new_name=paste0(Individual,&quot;_&quot;,new_trt,&quot;_&quot;,Time,&quot;_&quot;,sex,&quot;_&quot;,Replicate)) 

mapped &lt;- select(org.Hs.eg.db,
                 keys = row.names(raw_counts),
                 columns = c(&quot;SYMBOL&quot;),
                 keytype = &quot;ENTREZID&quot;)

mapped_unique_symbolref &lt;- mapped %&gt;%
  as.data.frame() %&gt;% 
  group_by(ENTREZID) %&gt;%
  dplyr::slice(1) %&gt;%
  ungroup()</code></pre>
<div id="initial-eval-of-matrix" class="section level3">
<h3>Initial eval of matrix</h3>
<pre class="r"><code>par(mfrow = c(1, 2))  # Set layout: 1 row, 2 columns
hist(raw_counts, main= &quot;Raw counts across samples&quot;,
     xlab = &quot;Raw counts&quot;,
     col=4)
hist(cpm(raw_counts, log=TRUE), 
     main = expression(&quot;Histogram of Log&quot;[2]*&quot; cpm unfiltered&quot;),
     xlab = expression(&quot;Log&quot;[2]*&quot; counts-per-million&quot;),
     col=4)</code></pre>
<p><img src="figure/Counts_matrix.Rmd/hist%20unfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1, 1))  # Reset layout to default after

boxplot(cpm(raw_counts, log=TRUE),
        main=expression(&quot;Boxplot of Log&quot;[2]*&quot; counts-per-million&quot;),
                        col=4,
        names=colnames(raw_counts),
        las=2, cex.axis=.5)</code></pre>
<p><img src="figure/Counts_matrix.Rmd/hist%20unfilt-2.png" width="960" style="display: block; margin: auto;" />
The raw count matrix has 28395 genes in ENTREZID or NCBI_gene
format.</p>
<pre class="r"><code>lcpm &lt;- cpm(raw_counts, log= TRUE)
  ### for determining the basic cutoffs
filt_raw_counts &lt;- raw_counts[rowMeans(lcpm)&gt; 0,]
dim(filt_raw_counts)</code></pre>
<pre><code>[1] 13977    84</code></pre>
<pre class="r"><code>#tail(rownames(filt_raw_counts),n=10)
filt_raw_counts_1 &lt;- raw_counts[rowMeans(lcpm)&gt; 1,]
filt_raw_counts_0.5 &lt;- raw_counts[rowMeans(lcpm)&gt; 0.5,]</code></pre>
<p>Filtering using row_means of the log2cpm &gt;0 yields 13977, 84,
genes for all columns. If I had used rowMeans &gt;0.5, I would have had
13384, 84 genes. If I had used rowMeans &gt;1, I would have 12837, 84
genes.</p>
<pre class="r"><code>par(mfrow = c(1, 2))  # Set layout: 1 row, 2 columns
hist(filt_raw_counts, main= &quot;Raw counts across samples&quot;,
     xlab = &quot;Raw counts&quot;,
     col=6)
hist(cpm(filt_raw_counts, log=TRUE), 
     main = expression(&quot;Histogram of Log&quot;[2]*&quot; cpm filtered rowMeans&gt;0&quot;),
     xlab = expression(&quot;Log&quot;[2]*&quot; counts-per-million&quot;),
     col=6)</code></pre>
<p><img src="figure/Counts_matrix.Rmd/hist%20filt-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1, 1))  # Reset layout to default after

boxplot(cpm(filt_raw_counts, log=TRUE),
        main=expression(&quot;Boxplot of Log&quot;[2]*&quot; counts-per-million of filtered set&quot;),
                        col=6,
        names=colnames(raw_counts),
        las=2, cex.axis=.5)</code></pre>
<p><img src="figure/Counts_matrix.Rmd/hist%20filt-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1, 2))  # Set layout: 1 row, 2 columns
hist(cpm(filt_raw_counts_0.5, log=TRUE), 
     main = expression(&quot;Histogram of Log&quot;[2]*&quot; cpm filtered rowMeans&gt;0.5&quot;),
     xlab = expression(&quot;Log&quot;[2]*&quot; counts-per-million&quot;),
     col=3)
hist(cpm(filt_raw_counts, log=TRUE), 
     main = expression(&quot;Histogram of Log&quot;[2]*&quot; cpm filtered rowMeans &gt;1&quot;),
     xlab = expression(&quot;Log&quot;[2]*&quot; counts-per-million&quot;),
     col=7)</code></pre>
<p><img src="figure/Counts_matrix.Rmd/hist%20filt-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1, 1))  # Reset layout to default after</code></pre>
<p>The distribution of log2cpm is fairly standard across all filtering
steps, so I am going with the less stringent filtering of rowMeans &gt;
0.</p>
</div>
<div id="heatmap-of-log2cpm-filtered-genes" class="section level3">
<h3>Heatmap of log<sub>2</sub>cpm filtered genes</h3>
<pre class="r"><code>## adding in renamed filt_raw_counts columns:
rename_vec &lt;- setNames(new_names$new_name, new_names$Core_ID)
filt_raw_counts &lt;- filt_raw_counts %&gt;% 
  as.data.frame() %&gt;% 
  rename_with(~ rename_vec[.x], .cols = all_of(new_names$Core_ID)) %&gt;% 
  as.matrix()</code></pre>
<pre class="r"><code>cor_lcpm_counts_pearson &lt;- filt_raw_counts %&gt;% 
  cpm(., log = TRUE) %&gt;% 
  cor(.,method = &quot;pearson&quot;)

cor_lcpm_counts_spearman &lt;- filt_raw_counts %&gt;% 
  cpm(., log = TRUE) %&gt;% 
  cor(.,method = &quot;spearman&quot;)


anno_raw_counts &lt;- data.frame(timeset = colnames(filt_raw_counts))

counts_corr_mat &lt;-
anno_raw_counts %&gt;%
  separate(timeset, into = c(&quot;indv&quot;,&quot;glucose&quot;,&quot;time&quot;,&quot;sex&quot;,&quot;rep&quot;), sep= &quot;_&quot;) %&gt;% 
  mutate(indv=factor(indv, levels= c( &quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;)),
         glucose=factor(glucose, levels=c(&quot;low&quot;, &quot;norm&quot;, &quot;osnorm&quot;, &quot;high&quot;)),
         time=factor(time, levels = c(&quot;2hr&quot; ,&quot;24hr&quot;, &quot;1week&quot;)),
         sex=factor(sex, levels=c(&quot;f&quot;,&quot;m&quot;)),
         rep=factor(rep, levels=c(&quot;A&quot;,&quot;B&quot;)))

###pulling color palette:
cb_palette &lt;- brewer.pal(6, &quot;Set1&quot;) 
dr_palette &lt;- brewer.pal(4,&quot;Dark2&quot;)

     
 mat_colors &lt;- list(
   indv= cb_palette,
   glucose=dr_palette,
   time=c(&quot;blue1&quot;,&quot;purple4&quot;,&quot;red4&quot;),
   sex=c(&quot;purple&quot;, &quot;orange&quot;),
   rep=c(&quot;grey&quot;,&quot;grey2&quot;))

names(mat_colors$indv)   &lt;- unique(counts_corr_mat$indv)
names(mat_colors$time) &lt;- unique(counts_corr_mat$time)
names(mat_colors$glucose) &lt;- unique(counts_corr_mat$glucose)
names(mat_colors$sex) &lt;- unique(counts_corr_mat$sex)
names(mat_colors$rep) &lt;- unique(counts_corr_mat$rep)


htanno_full_pearson &lt;-  ComplexHeatmap::HeatmapAnnotation(df = counts_corr_mat, col = mat_colors)
Heatmap(cor_lcpm_counts_pearson, 
        column_title=&quot;Filtered log2cpm pearson correlation&quot;,
        top_annotation = htanno_full_pearson
        )</code></pre>
<p><img src="figure/Counts_matrix.Rmd/filtered%20heatmap-1.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code>htanno_full_spearman &lt;-  ComplexHeatmap::HeatmapAnnotation(df = counts_corr_mat, col = mat_colors)
Heatmap(cor_lcpm_counts_spearman, 
        top_annotation = htanno_full_spearman,
        column_title=&quot;Filtered log2cpm spearman correlation&quot;)</code></pre>
<p><img src="figure/Counts_matrix.Rmd/filtered%20heatmap-2.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-analysis" class="section level3">
<h3>PCA analysis</h3>
<pre class="r"><code>prop_var_percent &lt;- function(pca_result){ 
  # Ensure the input is a PCA result object
  if (!inherits(pca_result, &quot;prcomp&quot;)) {
    stop(&quot;Input must be a result from prcomp()&quot;)
  }
  
  # Get the standard deviations from the PCA result
  sdev &lt;- pca_result$sdev
  
  # Calculate the proportion of variance
  proportion_variance &lt;- (sdev^2) / sum(sdev^2)*100
  
  return(proportion_variance)
}

get_regr_pval &lt;- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == &quot;lm&quot;)
  fstat &lt;- summary(mod)$fstatistic
  pval &lt;- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc &lt;- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char &lt;- paste0(&quot;PC&quot;, pc_num)
  # Calculate F-statistic p-value for linear model
  pval &lt;- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = &quot;lm&quot;) + labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  }
}</code></pre>
<pre class="r"><code>lcpm_filt_counts &lt;- filt_raw_counts %&gt;% 
  cpm(.,log = TRUE)
 
df_anno_mat &lt;- anno_raw_counts %&gt;%
  separate(timeset, into = c(&quot;indv&quot;,&quot;glucose&quot;,&quot;time&quot;,&quot;sex&quot;,&quot;rep&quot;), sep= &quot;_&quot;) %&gt;%
  mutate(indv=factor(indv, levels= c( &quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;)),
         glucose=factor(glucose, levels=c(&quot;low&quot;, &quot;norm&quot;, &quot;osnorm&quot;, &quot;high&quot;)),
         time=factor(time, levels = c(&quot;2hr&quot; ,&quot;24hr&quot;, &quot;1week&quot;)),
         sex=factor(sex, levels=c(&quot;f&quot;,&quot;m&quot;)),
         rep=factor(rep, levels=c(&quot;A&quot;,&quot;B&quot;)))
pca_lcpm_filt &lt;- prcomp(t(lcpm_filt_counts), scale. = TRUE)
pca_filt_anno &lt;- data.frame(df_anno_mat, pca_lcpm_filt$x)
plotting_var_names &lt;- prop_var_percent(pca_lcpm_filt)

pca_filt_anno %&gt;% 
ggplot(.,aes(x = PC1, y = PC2, col=glucose, shape=time, group=rep))+
  geom_point(size= 3)+
   ggrepel::geom_text_repel(aes(label = interaction(indv,rep)))+
   ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered peak set&quot;))+
  theme_bw()+
  guides(size =4)+
  labs(y = paste0(&quot;PC 2 (&quot;,round(plotting_var_names[2],2),&quot;%)&quot;)
       , x =paste0(&quot;PC 1 (&quot;,round(plotting_var_names[1],2),&quot;%)&quot;))+
  theme(plot.title=element_text(size= 14,hjust = 0.5),
        axis.title = element_text(size = 12, color = &quot;black&quot;))</code></pre>
<p><img src="figure/Counts_matrix.Rmd/pca%20plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_filt_anno %&gt;% 
ggplot(.,aes(x = PC3, y = PC4, col=glucose, shape=time, group=glucose))+
  geom_point(size= 3)+
   ggrepel::geom_text_repel(aes(label = interaction(indv,rep)))+
   ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered peak set&quot;))+
  theme_bw()+
  guides( size =4)+
  labs(y = paste0(&quot;PC 4 (&quot;,round(plotting_var_names[4],2),&quot;%)&quot;)
       , x =paste0(&quot;PC 3 (&quot;,round(plotting_var_names[3],2),&quot;%)&quot;))+
  theme(plot.title=element_text(size= 14,hjust = 0.5),
        axis.title = element_text(size = 12, color = &quot;black&quot;))</code></pre>
<p><img src="figure/Counts_matrix.Rmd/pca%20plots-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_filt_anno %&gt;% 
ggplot(.,aes(x = PC2, y = PC3, col=glucose, shape=time, group=glucose))+
  geom_point(size= 3)+
   ggrepel::geom_text_repel(aes(label = interaction(indv,rep)))+
   ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered peak set&quot;))+
  theme_bw()+
  guides( size =4)+
  labs(y = paste0(&quot;PC 3 (&quot;,round(plotting_var_names[3],2),&quot;%)&quot;)
       , x =paste0(&quot;PC 2 (&quot;,round(plotting_var_names[2],2),&quot;%)&quot;))+
  theme(plot.title=element_text(size= 14,hjust = 0.5),
        axis.title = element_text(size = 12, color = &quot;black&quot;))</code></pre>
<p><img src="figure/Counts_matrix.Rmd/pca%20plots-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="contribution-to-variance" class="section level3">
<h3>Contribution to variance</h3>
<pre class="r"><code>facs &lt;- c(&quot;indv&quot;, &quot;glucose&quot;, &quot;time&quot;, &quot;sex&quot;)
names(facs) &lt;- c(&quot;Individual&quot;, &quot;Glucose level&quot;, &quot;Time&quot;, &quot;Sex&quot;)

for (f in facs) {
  # PC1 v PC2
  # pca_plot(pca_final_four_anno, col_var = f, shape_var = time,
  #          title = names(facs)[which(facs == f)])
  # print(last_plot())
  
  # Plot f versus PC1 and PC2
  f_v_pc1 &lt;- gridExtra::arrangeGrob(plot_versus_pc(pca_filt_anno, 1, f))
  f_v_pc2 &lt;- gridExtra::arrangeGrob(plot_versus_pc(pca_filt_anno, 2, f))
  f_v_pc3 &lt;- gridExtra::arrangeGrob(plot_versus_pc(pca_filt_anno, 3, f))
  f_v_pc4 &lt;- gridExtra::arrangeGrob(plot_versus_pc(pca_filt_anno, 4, f))
  f_v_pc5 &lt;- gridExtra::arrangeGrob(plot_versus_pc(pca_filt_anno, 5, f))
  gridExtra::grid.arrange(f_v_pc1, f_v_pc2, f_v_pc3,f_v_pc4, f_v_pc5, ncol = 3, nrow=2, top = names(facs)[which(facs == f)])
}</code></pre>
<p><img src="figure/Counts_matrix.Rmd/variance%20by%20type-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Counts_matrix.Rmd/variance%20by%20type-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Counts_matrix.Rmd/variance%20by%20type-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Counts_matrix.Rmd/variance%20by%20type-4.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggpubr_0.6.0                            
 [2] RColorBrewer_1.1-3                      
 [3] TxDb.Hsapiens.UCSC.hg38.knownGene_3.20.0
 [4] GenomicFeatures_1.58.0                  
 [5] org.Hs.eg.db_3.20.0                     
 [6] AnnotationDbi_1.68.0                    
 [7] Biobase_2.66.0                          
 [8] readxl_1.4.5                            
 [9] GenomicRanges_1.58.0                    
[10] GenomeInfoDb_1.42.3                     
[11] IRanges_2.40.1                          
[12] S4Vectors_0.44.0                        
[13] BiocGenerics_0.52.0                     
[14] genomation_1.38.0                       
[15] data.table_1.17.0                       
[16] ComplexHeatmap_2.22.0                   
[17] edgeR_4.4.2                             
[18] limma_3.62.2                            
[19] DT_0.33                                 
[20] kableExtra_1.4.0                        
[21] lubridate_1.9.4                         
[22] forcats_1.0.0                           
[23] stringr_1.5.1                           
[24] dplyr_1.1.4                             
[25] purrr_1.0.4                             
[26] readr_2.1.5                             
[27] tidyr_1.3.1                             
[28] tibble_3.2.1                            
[29] ggplot2_3.5.1                           
[30] tidyverse_2.0.0                         
[31] workflowr_1.7.1                         

loaded via a namespace (and not attached):
  [1] rstudioapi_0.17.1           jsonlite_1.9.1             
  [3] shape_1.4.6.1               magrittr_2.0.3             
  [5] magick_2.8.5                farver_2.1.2               
  [7] rmarkdown_2.29              GlobalOptions_0.1.2        
  [9] fs_1.6.5                    BiocIO_1.16.0              
 [11] zlibbioc_1.52.0             vctrs_0.6.5                
 [13] Cairo_1.6-2                 memoise_2.0.1              
 [15] Rsamtools_2.22.0            RCurl_1.98-1.16            
 [17] rstatix_0.7.2               htmltools_0.5.8.1          
 [19] S4Arrays_1.6.0              plotrix_3.8-4              
 [21] curl_6.2.1                  broom_1.0.7                
 [23] cellranger_1.1.0            Formula_1.2-5              
 [25] SparseArray_1.6.2           sass_0.4.9                 
 [27] KernSmooth_2.23-26          bslib_0.9.0                
 [29] htmlwidgets_1.6.4           plyr_1.8.9                 
 [31] impute_1.80.0               cachem_1.1.0               
 [33] GenomicAlignments_1.42.0    whisker_0.4.1              
 [35] lifecycle_1.0.4             iterators_1.0.14           
 [37] pkgconfig_2.0.3             Matrix_1.7-3               
 [39] R6_2.6.1                    fastmap_1.2.0              
 [41] GenomeInfoDbData_1.2.13     MatrixGenerics_1.18.1      
 [43] clue_0.3-66                 digest_0.6.37              
 [45] colorspace_2.1-1            ps_1.9.0                   
 [47] rprojroot_2.0.4             RSQLite_2.3.9              
 [49] labeling_0.4.3              timechange_0.3.0           
 [51] httr_1.4.7                  abind_1.4-8                
 [53] compiler_4.4.2              bit64_4.6.0-1              
 [55] withr_3.0.2                 doParallel_1.0.17          
 [57] backports_1.5.0             BiocParallel_1.40.0        
 [59] carData_3.0-5               DBI_1.2.3                  
 [61] ggsignif_0.6.4              DelayedArray_0.32.0        
 [63] rjson_0.2.23                tools_4.4.2                
 [65] httpuv_1.6.15               glue_1.8.0                 
 [67] restfulr_0.0.15             callr_3.7.6                
 [69] promises_1.3.2              getPass_0.2-4              
 [71] gridBase_0.4-7              cluster_2.1.8.1            
 [73] reshape2_1.4.4              generics_0.1.3             
 [75] gtable_0.3.6                BSgenome_1.74.0            
 [77] tzdb_0.4.0                  seqPattern_1.38.0          
 [79] hms_1.1.3                   car_3.1-3                  
 [81] xml2_1.3.7                  XVector_0.46.0             
 [83] ggrepel_0.9.6               foreach_1.5.2              
 [85] pillar_1.10.1               vroom_1.6.5                
 [87] later_1.4.1                 circlize_0.4.16            
 [89] lattice_0.22-6              bit_4.6.0                  
 [91] rtracklayer_1.66.0          tidyselect_1.2.1           
 [93] locfit_1.5-9.12             Biostrings_2.74.1          
 [95] knitr_1.49                  git2r_0.35.0               
 [97] gridExtra_2.3               SummarizedExperiment_1.36.0
 [99] svglite_2.1.3               xfun_0.51                  
[101] statmod_1.5.0               matrixStats_1.5.0          
[103] stringi_1.8.4               UCSC.utils_1.2.0           
[105] yaml_2.3.10                 evaluate_1.0.3             
[107] codetools_0.2-20            cli_3.6.4                  
[109] systemfonts_1.2.1           munsell_0.5.1              
[111] processx_3.8.6              jquerylib_0.1.4            
[113] Rcpp_1.0.14                 png_0.1-8                  
[115] XML_3.99-0.18               parallel_4.4.2             
[117] blob_1.2.4                  bitops_1.0-9               
[119] viridisLite_0.4.2           scales_1.3.0               
[121] crayon_1.5.3                GetoptLong_1.0.5           
[123] rlang_1.1.5                 KEGGREST_1.46.0            </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
