---
title: "Fastqc_initial"
author: "Renee Matthews"
date: "2025-05-26"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
			dev = c("png","pdf")
	
)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```

```{r package loading}
library(tidyverse)
library(readr)
library(kableExtra)
library(DT)
library(readr)
library(edgeR)
# library(ComplexHeatmap)
library(data.table)
# library(genomation)
# library(GenomicRanges)
library(readxl)
```


##### Loading data
```{r Creating annodata file, eval=FALSE}
NGS_RNA_Samples_OJ <- read_excel("C:/Users/renee/Other_projects_data/Glucose_files/NGS_RNA_Samples_OJ.xlsx")
sample_info <- read_excel("C:/Users/renee/Other_projects_data/Glucose_files/Version5_Glucose_Samples_RNA_extraction_OJ_240905.xlsx")

organized_sample_info <- sample_info %>% 
  dplyr::select(Sample_ID,Individual,Treatment:Time,Replicate,Line,'Batch#') %>% 
  mutate(Replicate=case_match(Replicate,1~"A",
                              2~"B",
                             .default = "Try again")) %>% 
  left_join(., (NGS_RNA_Samples_OJ %>% 
                  dplyr::rename("Core_ID"="Sample_ID")),
            c("Sample_ID"="Sample_name")) %>% 
  dplyr::select(Sample_ID:Core_ID) %>% 
  write_tsv(.,"data/Annotation_data/organized_sample_info.tsv")
```

```{r fastqc files initial}
multiqc_fastqc <- read_delim("C:/Users/renee/Other_projects_data/Glucose_files/fastq_files/Ward_Run1113/multiqc_files/multiqc_data/multiqc_fastqc.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
multiqc_fastqc_genstat <- read_delim("C:/Users/renee/Other_projects_data/Glucose_files/fastq_files/Ward_Run1113/multiqc_files/multiqc_data/multiqc_general_stats.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

multiqc_fastqc %>% 
  left_join(., multiqc_fastqc_genstat) %>% 
  write_tsv(.,"data/Fastqc_data/Fastqc_multiqc_report_RUN1113.tsv")
##############################################
multiqc_fastqc1 <- read_delim("C:/Users/renee/Other_projects_data/Glucose_files/fastq_files/Ward_Run1144_RepeatRun1113/multiqc_files/multiqc_data/multiqc_fastqc.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
multiqc_fastqc_genstat1 <- read_delim("C:/Users/renee/Other_projects_data/Glucose_files/fastq_files/Ward_Run1144_RepeatRun1113/multiqc_files/multiqc_data/multiqc_general_stats.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

multiqc_fastqc1 %>% 
  left_join(., multiqc_fastqc_genstat1) %>% 
  write_tsv(.,"data/Fastqc_data/Fastqc_multiqc_report_RUN1144.tsv")


```
### Reads by sample
```{r reads by sample, fig.width=12}
MQC_1113 <- multiqc_fastqc %>% 
  left_join(., multiqc_fastqc_genstat) %>% 
  extract(., Sample, into = c("prefix","read"), regex= "(.+)_R(\\d+)", remove=FALSE) %>% 
  mutate(read = paste0("R", read))

MQC_1144 <- multiqc_fastqc1 %>% 
  left_join(., multiqc_fastqc_genstat1) %>% 
  extract(., Sample, into = c("prefix","read"), regex= "(.+)_R(\\d+)", remove=FALSE) %>% 
  mutate(read = paste0("R", read))

MQC_1113 %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x=Sample,y=`Total Sequences`))+
  geom_col(position="dodge",aes()) + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("count")+
  ggtitle("Run 1113, Read 1 sequences")
MQC_1113 %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x=Sample,y=`Total Sequences`))+
  geom_col(position="dodge",aes()) + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("count")+
  ggtitle("Run 1113, Read 2 sequences")

MQC_1144 %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x=Sample,y=`Total Sequences`))+
  geom_col(position="dodge",aes()) + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("count")+
  ggtitle("Run 1144, Read 1 sequences")
MQC_1144 %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x=Sample,y=`Total Sequences`))+
  geom_col(position="dodge",aes()) + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("count")+
  ggtitle("Run 1144, Read 2 sequences")

```

### Percent duplication


```{r percent dups, fig.width=12}

MQC_1113 %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x=Sample,y=`FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(position="dodge",fill="purple1") + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("percent duplication")+
  ggtitle("Run 1113, Read 1 sequences, percent duplicates")
MQC_1113 %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x=Sample,y=`FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(position="dodge",fill="purple1") + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("percent duplication")+
  ggtitle("Run 1113, Read 2 sequences, percent duplicates")

MQC_1144 %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x=Sample,y=`FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(position="dodge",fill="cornflowerblue") + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("percent duplication")+
  ggtitle("Run 1144, Read 1 sequences, percent duplicates")
MQC_1144 %>% 
  dplyr::filter(read=="R2") %>% 
 ggplot(., aes(x=Sample,y=`FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(position="dodge",fill="cornflowerblue") + 
  theme(axis.text.x=element_text(vjust=.2,angle=90)) +
  ylab("percent duplication")+
  ggtitle("Run 1144, Read 2 sequences, percent duplicates")


```
#### loading concatenated data

```{r concat loading}
multiqc_fastqc_concat <- read_delim("C:/Users/renee/Other_projects_data/Glucose_files/concatenated_fastqc/multiqc_summary/multiqc_data/multiqc_fastqc.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
multiqc_fastqc_genstat_concat <- read_delim("C:/Users/renee/Other_projects_data/Glucose_files/concatenated_fastqc/multiqc_summary/multiqc_data/multiqc_general_stats.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
sample_info_concat <- read_tsv("data/Annotation_data/organized_sample_info.tsv")
```

```{r}
MQC_concat <-
multiqc_fastqc_concat %>% 
  left_join(., multiqc_fastqc_genstat_concat) %>% 
  extract(., Sample, into = c("prefix","read"), regex= "(.+)_R(\\d+)", remove=FALSE) %>% 
  mutate(read = paste0("R", read)) %>% 
  left_join(., sample_info_concat, by=c("prefix"="Core_ID"))

plot_dataframe <-MQC_concat %>% 
  dplyr::select(prefix, read, Sample,'Total Sequences',total_deduplicated_percentage,`FastQC_mqc-generalstats-fastqc-percent_duplicates`:`FastQC_mqc-generalstats-fastqc-total_sequences`,Sample_ID:Line) %>% 
  dplyr::rename("Conc"=`Conc(mM)`) %>% 
  rename_with(
    .cols = starts_with("FastQC_mqc-generalstats-fastqc-"),
    .fn = ~ str_replace(., "FastQC_mqc-generalstats-fastqc-", "")
  ) %>% 
  mutate(Time=factor(Time, levels = c("2hr","24hr","1week")),
         Conc = factor(Conc, levels= c("0","5","5+25","30")))

```

### Total sequences across samples
```{r tot-seq concat, fig.width=14}
plot_dataframe %>% 
dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x=prefix,y=`Total Sequences`,fill=Treatment))+
  geom_col(position="dodge") + 
  geom_text(aes(label = Sample_ID),
            position = position_stack(vjust = 0.5),
            angle    = 90) +
   scale_y_continuous(expand = c(0, 0))+
  theme_minimal()+
  theme(axis.text.x=element_text(vjust=.2,angle=90),
        legend.position = "bottom") +
  ylab("count")+
  ggtitle("Fastqc Concatenated, Total read count")
```

### Total duplication across samples
```{r tot-dup concat, fig.width=14}

plot_dataframe %>% 
dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x=prefix,y=percent_duplicates,fill=Treatment))+
  geom_col(position="dodge") + 
  geom_text(aes(label = Sample_ID),
            position = position_stack(vjust = 0.5),
            angle    = 90) +
   scale_y_continuous(expand = c(0, 0))+
  theme_minimal()+
  theme(axis.text.x=element_text(vjust=.2,angle=90),
        legend.position = "bottom") +
  ylab("count")+
  ggtitle("Fastqc Concatenated, duplication amount Read 1")


plot_dataframe %>% 
dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x=prefix,y=percent_duplicates,fill=Treatment))+
  geom_col(position="dodge") + 
  geom_text(aes(label = Sample_ID),
            position = position_stack(vjust = 0.5),
            angle    = 90) +
   scale_y_continuous(expand = c(0, 0))+
  theme_minimal()+
  theme(axis.text.x=element_text(vjust=.2,angle=90),
        legend.position = "bottom") +
  ylab("count")+
  ggtitle("Fastqc Concatenated, duplication amount Read 2")
```
### Boxplot of counts by time, treatment, individual, concentration


```{r Boxplots}
plot_dataframe %>% 
  dplyr::filter(read=="R1") %>% 
    ggplot(., aes(x=Treatment,y=`Total Sequences`,fill=Treatment))+
  geom_boxplot()+
  theme_minimal()+
  ggtitle("Counts across treatments")

plot_dataframe %>% 
  dplyr::filter(read=="R1")%>% 
    ggplot(., aes(x=Line,y=`Total Sequences`, fill = Line))+
  geom_boxplot()+
  theme_minimal()+
  ggtitle("Counts across individual")

plot_dataframe %>% 
  dplyr::filter(read=="R1")%>% 
  ggplot(., aes(x=Time,y=`Total Sequences`, fill = Time))+
  geom_boxplot()+
  theme_minimal()+
  ggtitle("Counts across time")

plot_dataframe %>% 
  dplyr::filter(read=="R1")%>% 
  ggplot(., aes(x=Conc,y=`Total Sequences`, fill = Conc))+
  geom_boxplot()+
  theme_minimal()+
  ggtitle("Counts across concentration")


```

#### looking at replicate and original (75-1)
```{r}
plot_dataframe %>% 
  dplyr::filter(read=="R1"& Individual=="1") %>% 
    ggplot(., aes(x=Replicate,y=`Total Sequences`,fill=Treatment))+
  geom_boxplot()+
  geom_point(position = position_dodge(width = 0.9), 
             shape = 21, size = 2, stroke = 0.5, aes(fill = Treatment))+
  theme_minimal()+
  ggtitle("Counts across treatments")


plot_dataframe %>% 
  dplyr::filter(read=="R1"& Individual=="1") %>% 
    ggplot(., aes(x=Replicate,y=`Total Sequences`,fill=Time))+
  geom_boxplot()+
  geom_point(position = position_dodge(width = 0.9), 
             shape = 21, size = 2, stroke = 0.5, aes(fill = Time))+
  theme_minimal()+
  ggtitle("Counts across treatments")


```
